service/dap/server.go

```diff
@@ -882,22 +882,27 @@ func (s *Server) onStepOutRequest(request *dap.StepOutRequest) {
 
 func (s *Server) doStepCommand(command string, threadId int) {
 	// Use SwitchGoroutine to change the current goroutine.
-	state, err := s.debugger.Command(&api.DebuggerCommand{Name: api.SwitchGoroutine, GoroutineID: threadId}, nil)
-	if err != nil {
-		s.log.Errorf("Error switching goroutines while stepping: %e", err)
-		// If we encounter an error, we will have to send a stopped event
-		// since we already sent the step response.
-		stopped := &dap.StoppedEvent{Event: *newEvent("stopped")}
-		stopped.Body.AllThreadsStopped = true
-		if state.SelectedGoroutine != nil {
-			stopped.Body.ThreadId = state.SelectedGoroutine.ID
-		} else if state.CurrentThread != nil {
-			stopped.Body.ThreadId = state.CurrentThread.GoroutineID
+	var err error
+	if threadId != 0 {
```

Comment:
So what is the meaning of this check? That we assume that value of 0 means it was not set because delve would never return 0 as a meaningful value? Why would we not just handle 0 as any other value?

DAP spec includes required thread ids for all running requests. For continue they say if you must continue on all threads, then just reflect that in the response, but for next and step in/out they assume that it is possible to only advance on that one thread. Then UI then marks the one thread where you are stepping as running and the rest as paused with UI options to step on them.

Reply:
> Why would we not just handle 0 as any other value?

Removed the check.